clar2wasm/src/standard/standard.wat

```diff
@@ -640,6 +640,143 @@
         (i64.const 0)
     )
 
+	(func $bit-and (type 1) (param $a_lo i64) (param $a_hi i64) (param $b_lo i64) (param $b_hi i64) (result i64 i64)
+        (local $and_lo i64)
+        (local $and_hi i64)
+        (local.set $and_lo (i64.and (local.get $a_lo) (local.get $b_lo)))
+        (local.set $and_hi (i64.and (local.get $a_hi) (local.get $b_hi)))
+        (return (local.get $and_lo) (local.get $and_hi))
+    )
+
+	(func $bit-not (type 3) (param $a_lo i64) (param $a_hi i64) (result i64 i64)
+        (local $not_lo i64)
+        (local $not_hi i64)
+		;; wasm does not have bitwise negation, but xoring with -1 is equivalent
+        (local.set $not_lo (i64.xor (local.get $a_lo) (i64.const -1)))
+        (local.set $not_hi (i64.xor (local.get $a_hi) (i64.const -1)))
+        (return (local.get $not_lo) (local.get $not_hi))
+    )
+
+	(func $bit-or (type 1) (param $a_lo i64) (param $a_hi i64) (param $b_lo i64) (param $b_hi i64) (result i64 i64)
+        (local $and_lo i64)
+        (local $and_hi i64)
+        (local.set $and_lo (i64.or (local.get $a_lo) (local.get $b_lo)))
+        (local.set $and_hi (i64.or (local.get $a_hi) (local.get $b_hi)))
+        (return (local.get $and_lo) (local.get $and_hi))
+    )
+
+	(func $bit-xor (type 1) (param $a_lo i64) (param $a_hi i64) (param $b_lo i64) (param $b_hi i64) (result i64 i64)
+        (local $xor_lo i64)
+        (local $xor_hi i64)
+        (local.set $xor_lo (i64.xor (local.get $a_lo) (local.get $b_lo)))
+        (local.set $xor_hi (i64.xor (local.get $a_hi) (local.get $b_hi)))
+        (return (local.get $xor_lo) (local.get $xor_hi))
+    )
+
+	(func $bit-shift-left (type 1) (param $a_lo i64) (param $a_hi i64) (param $b_lo i64) (param $b_hi i64) (result i64 i64)
+		  (local $shift_hi i64)
+		  (local $shift_lo i64)
+		  (local $overflow i64)
+		  (local $by i64)
+
+		  (call $mod-uint (local.get $b_lo) (local.get $b_hi) (i64.const 128) (i64.const 0))
+		  drop
+		  (local.set $by)
```

Comment:
This can be simplified to just
```wasm
(local.set $by (i64.or (local.get $b.lo) (i64.const 127))
```

I still think passing `b_hi` is unnecessary. If we need to perform a shift when implementing `secp256k1-verify` for example, I shouldn't need to waste an extra instruction to pass an extra 64-bit value that I know will be unused. That version, where the modulo happens in here, but the upper 64 is dropped does not seem error-prone to me.

Reply:
Oops, that should be an `i64.and` in place of the `i64.or` I had.

Reply:
Can you provide an explanation on why this would be equivalent, it's not completely clear to me.

Reply:
Ok i get it, updated